frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/useServingRuntimeConfig.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { ServingRuntimeKind } from 'k8sTypes';
+import { fetchServingRuntimeConfig } from 'api';
+import { useDashboardNamespace } from 'redux/selectors';
+
+const useServingRuntimeConfig = (
+  open: boolean,
+): [
+  servingRuntimeConfig: ServingRuntimeKind | null,
+  loaded: boolean,
+  loadError: Error | undefined,
+] => {
+  const { dashboardNamespace } = useDashboardNamespace();
+  const [servingRuntimeConfig, setServingRuntimeConfig] = React.useState<ServingRuntimeKind | null>(
+    null,
+  );
+  const [loaded, setLoaded] = React.useState(false);
+  const [loadError, setLoadError] = React.useState<Error | undefined>(undefined);
+
+  const getServingRuntimeConfig = React.useCallback(() => {
+    setLoaded(false);
```

Comment:
Nit. This is a bad pattern to reset loading or data before fetching.

We really need to get that core utility together, it'll fix this -- you're throwing away data before you fetch -- it could go "here is information" and at a random point to the user it'll flip to "loading information" and then back to "here is information" in a fast succession (most likely). This leads to a wonky UX because the data should not be respected while it is loading (ie the code using this should check loading first before data). Personally I do not like this model as it typically leads to flickers in the UI.

However -- this is a nit because your items you are waiting on are never going to change. `open` state, `dashboard namespace` -- these are not going to change on the user. 

FWIW, I can guess (you can confirm) why you're doing this -- and that's for each model usage, which should not be treated alongside the fetch function. You sorta wanna do a `useEffect => setLoaded(!open)` sorta logic.